cpr/session.cpp

```diff
@@ -24,6 +24,7 @@ constexpr long OFF = 0L;
 class Session::Impl {
   public:
     Impl();
+    ~Impl();
```

Comment:
Any specific reason why `~Impl() = default;` is not working here?

Reply:
I did not try `~Impl() = default;` i think  would be fine. 

- to fix this bug properly `Impl` class required a custom deleter required.

**reason**:std::unique_ptr The size of the type needs to be statically detected in static_assert(sizeof(Impl)>0, But our Impl Is a pre declared type.  It's impossible to calculate , So it leads to an error.
I create a commit to fix.